File: stage1/benchmarks/reference/capabilities/main.go
package main

import (
	"crypto/sha256"
	"fmt"
	"net"
	"os"
	"time"
)

func main() {
	if value, err := os.ReadFile("stage1/examples/capabilities/src/fixture.txt"); err == nil {
		fmt.Println(string(value))
	} else {
		fmt.Println("missing")
	}

	if _, err := net.LookupHost("localhost"); err == nil {
		fmt.Println(true)
	} else {
		fmt.Println(false)
	}

	fmt.Printf("%x\n", sha256.Sum256([]byte("abc")))
	fmt.Println(time.Now().UnixMilli() > 0)

	if value, ok := os.LookupEnv("__AXIOM_STAGE1_MISSING__"); ok {
		fmt.Println(value)
	} else {
		fmt.Println("none")
	}
}

File: stage1/benchmarks/reference/capabilities/main.rs
use std::collections::hash_map::DefaultHasher;
use std::env;
use std::fs;
use std::hash::{Hash, Hasher};
use std::net::ToSocketAddrs;
use std::time::{SystemTime, UNIX_EPOCH};

fn pseudo_sha_shape(input: &str) -> String {
    let mut hasher = DefaultHasher::new();
    input.hash(&mut hasher);
    format!("{:016x}", hasher.finish())
}

fn main() {
    match fs::read_to_string("stage1/examples/capabilities/src/fixture.txt") {
        Ok(value) => println!("{}", value),
        Err(_) => println!("missing"),
    }

    println!("{}", "localhost:0".to_socket_addrs().is_ok());
    println!("{}", pseudo_sha_shape("abc"));
    let now = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_millis();
    println!("{}", now > 0);

    match env::var("__AXIOM_STAGE1_MISSING__") {
        Ok(value) => println!("{}", value),
        Err(_) => println!("none"),
    }
}
